<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>try catch里面try catch嵌套 - IT学习者博客</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
		<meta property="og:title" content="try catch里面try catch嵌套" />
<meta property="og:description" content=" try catch里面try catch嵌套
点击打开链接
try 与catch的作用 首先要清楚，如果没有try的话，出现异常会导致程序崩溃。而try则可以保证程序的正常运行下去，比如说：　try{ int i = 1/0; }catch(Exception e){ e.printStackTrace(); } 一个计算的话，如果除数为0，则会报错，如果没有try的话，程序直接崩溃。用try的话，则可以让程序运行下去，并且输出为什么出错！try的话，配合log4j使用会对程序的日后维护帮助很大。 那么e.printStackTrace();是什么意思？ 当try语句中出现异常是时，会执行catch中的语句，java运行时系统会自动将catch括号中的Exception e 初始化，也就是实例化Exception类型的对象。e是此对象引用名称。然后e（引用）会自动调用Exception类中指定的方法，也就出现了e.printStackTrace();。printStackTrace()方法的意思是：在命令行打印异常信息在程序中出错的位置及原因。（这是白话解释，比较容易理解） try{//代码区}catch(Exception e){//异常处理}代码区如果有错误，就会返回所写异常的处理。 " />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://itnewbiea.github.io/posts/ac2234a715d3cfe496c4b2249dd52a49/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2018-05-18T13:26:54+08:00" />
<meta property="article:modified_time" content="2018-05-18T13:26:54+08:00" />


	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="IT学习者博客" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">IT学习者博客</div>
					
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">try catch里面try catch嵌套</h1>
			
		</header>
		<div class="content post__content clearfix">
			
<div id="content_views" class="htmledit_views">
                    <p><a href="https://www.cnblogs.com/zhangchengye/p/5674240.html" rel="nofollow">try catch里面try catch嵌套</a></p> 
<p><br></p> 
<p><a href="https://www.cnblogs.com/liushao/p/6380613.html" rel="nofollow">点击打开链接</a><br></p> 
<p></p> 
<h2 class="postTitle" style="margin-top:0px;margin-bottom:15px;padding:0px 0px 0px 5px;font-size:15.6px;float:left;line-height:1.5em;width:1067.64px;clear:both;color:rgb(70,70,70);font-family:verdana, 'ms song', Arial, Helvetica, sans-serif;text-align:left;background-color:rgb(250,247,239);"><a class="postTitle2" href="http://www.cnblogs.com/liushao/p/6380613.html" rel="nofollow" style="margin:0px;padding:0px;color:rgb(100,102,179);">try 与catch的作用</a></h2> 
<div class="clear" style="margin:0px;padding:0px;clear:both;color:rgb(70,70,70);font-family:verdana, 'ms song', Arial, Helvetica, sans-serif;font-size:12px;text-align:left;background-color:rgb(250,247,239);"></div> 
<div class="postBody" style="margin:0px;padding:5px 2px 5px 5px;line-height:1.5;color:rgb(57,57,57);border-bottom:1px solid #000000;font-size:14px;font-family:verdana, 'ms song', Arial, Helvetica, sans-serif;text-align:left;background-color:rgb(250,247,239);"> 
 <div class="blogpost-body" style="margin:0px 0px 20px;padding:0px;"> 
  <pre class="best-text mb-10" style="margin-bottom:0px;padding-right:0px;padding-left:0px;white-space:pre-wrap;">首先要清楚，如果没有try的话，出现异常会导致程序崩溃。<br style="margin:0px;padding:0px;">而try则可以保证程序的正常运行下去，比如说：　</pre> 
  <div class="cnblogs_code" style="margin:5px 0px;padding:5px;background-color:rgb(245,245,245);border:1px solid rgb(204,204,204);color:rgb(0,0,0);font-family:'Courier New';font-size:12px;"> 
   <pre style="margin-bottom:0px;padding-right:0px;padding-left:0px;white-space:pre-wrap;font-family:'Courier New';"><span style="margin:0px;padding:0px;color:rgb(0,0,255);line-height:1.5;">try</span><span style="margin:0px;padding:0px;line-height:1.5;">{
　　</span><span style="margin:0px;padding:0px;color:rgb(0,0,255);line-height:1.5;">int</span> i = 1/0<span style="margin:0px;padding:0px;line-height:1.5;">;
}</span><span style="margin:0px;padding:0px;color:rgb(0,0,255);line-height:1.5;">catch</span><span style="margin:0px;padding:0px;line-height:1.5;">(Exception e){
    e.printStackTrace();
}</span></pre> 
  </div> 
  <pre class="best-text mb-10" style="margin-bottom:0px;padding-right:0px;padding-left:0px;white-space:pre-wrap;">　　一个计算的话，如果<a class="baidu-highlight" href="https://www.baidu.com/s?wd=%E9%99%A4%E6%95%B0%E4%B8%BA0&amp;tn=44039180_cpr&amp;fenlei=mv6quAkxTZn0IZRqIHckPjm4nH00T1Y3uj6kuA7buHu9mWnkmWwW0ZwV5Hcvrjm3rH6sPfKWUMw85HfYnjn4nH6sgvPsT6KdThsqpZwYTjCEQLGCpyw9Uz4Bmy-bIi4WUvYETgN-TLwGUv3EnWb3nW0dPW6L" rel="nofollow" style="margin:0px;padding:0px;color:rgb(100,102,179);">除数为0</a>，则会报错，如果没有try的话，程序直接崩溃。用try的话，则可以让程序运行下去，并且输出为什么出错！<br style="margin:0px;padding:0px;">try的话，配合log4j使用会对程序的日后维护帮助很大。</pre> 
  <pre style="margin-bottom:0px;padding-right:0px;padding-left:0px;white-space:pre-wrap;"><span style="margin:0px;padding:0px;">　　<span style="margin:0px;padding:0px;">那么e.printStackTrace();是什么意思？</span><br style="margin:0px;padding:0px;"></span></pre> 
  <pre class="best-text mb-10" style="margin-bottom:0px;padding-right:0px;padding-left:0px;white-space:pre-wrap;">　　当try语句中出现异常是时，会执行catch中的语句，java运行时系统会自动将catch括号中的Exception e 初始化，也就是实例化Exception类型的对象。e是此<a class="baidu-highlight" href="https://www.baidu.com/s?wd=%E5%AF%B9%E8%B1%A1%E5%BC%95%E7%94%A8&amp;tn=44039180_cpr&amp;fenlei=mv6quAkxTZn0IZRqIHckPjm4nH00T1YvPAcsuycYrjwbuWbzPAu-0ZwV5Hcvrjm3rH6sPfKWUMw85HfYnjn4nH6sgvPsT6KdThsqpZwYTjCEQLGCpyw9Uz4Bmy-bIi4WUvYETgN-TLwGUv3EPH04nWRsrjm" rel="nofollow" style="margin:0px;padding:0px;color:rgb(100,102,179);">对象引用</a>名称。然后e（引用）会自动调用Exception类中指定的方法，也就出现了e.printStackTrace();。<br style="margin:0px;padding:0px;">printStackTrace()方法的意思是：<span style="margin:0px;padding:0px;">在命令行打印异常信息在程序中出错的位置及原因</span>。（这是白话解释，比较容易理解）</pre> 
  <pre class="best-text mb-10" style="margin-bottom:0px;padding-right:0px;padding-left:0px;white-space:pre-wrap;"><br style="margin:0px;padding:0px;">try{<!-- --><br style="margin:0px;padding:0px;">//代码区<br style="margin:0px;padding:0px;">}catch(Exception e){<!-- --><br style="margin:0px;padding:0px;">//异常处理<br style="margin:0px;padding:0px;">}<br style="margin:0px;padding:0px;">代码区如果有错误，就会返回所写异常的处理。</pre> 
 </div> 
</div> 
<br>
                </div>
		</div>
	</article>
</main>


<nav class="pager flex">
	<div class="pager__item pager__item--prev">
		<a class="pager__link" href="/posts/e4a60ae011e32e35b13db3e9c336d8bd/" rel="prev">
			<span class="pager__subtitle">«&thinsp;Previous</span>
			<p class="pager__title">C#中事件（event）和委托（Delegate）的一些略解</p>
		</a>
	</div>
	<div class="pager__item pager__item--next">
		<a class="pager__link" href="/posts/95e4945942a41f0d1aed96e95aca971e/" rel="next">
			<span class="pager__subtitle">Next&thinsp;»</span>
			<p class="pager__title">android FDE功能介绍</p>
		</a>
	</div>
</nav>


			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2024 IT学习者博客.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>
<script src="https://itnewbiea.github.io/js/foot.js"></script>


	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>